fe/fe-core/src/main/java/org/apache/doris/qe/cache/SqlCache.java

```diff
@@ -35,13 +35,21 @@ public SqlCache(TUniqueId queryId, SelectStmt selectStmt) {
         super(queryId, selectStmt);
     }
 
+    public SqlCache(TUniqueId queryId) {
+        super(queryId);
+    }
+
     public void setCacheInfo(CacheAnalyzer.CacheTable latestTable, String allViewExpandStmtListStr) {
         this.latestTable = latestTable;
         this.allViewExpandStmtListStr = allViewExpandStmtListStr;
     }
 
     public String getSqlWithViewStmt() {
-        return selectStmt.toSql() + "|" + allViewExpandStmtListStr;
+        if (selectStmt != null)  {
+            return selectStmt.toSql() + "|" + allViewExpandStmtListStr;
```

Comment:
what about fold constant? For example, select * from tablet where a = cur_date(); although the

Reply:
If there is a random function, the result of sql cache will be wrong. This cannot be solved temporarily and needs to be decided by the user.